doc/whatsnew/fragments/8453.bugfix

```diff
@@ -0,0 +1,3 @@
+Fix incorrect preferred-modules matches.
```

Comment:
```suggestion
Fix a regression of ``preferred-modules`` where a partial match was used instead of the required full match.
```

Reply:
Thanks, this is way better! Adding this into the commits.